storage/src/journal/variable.rs

```diff
@@ -626,6 +626,88 @@ impl<E: Storage + Metrics, V: Codec> Journal<E, V> {
         blob.sync().await.map_err(Error::Runtime)
     }
 
+    /// Gets the size of a specific section in bytes.
+    ///
+    /// Returns 0 if the section does not exist.
+    pub async fn section_size(&self, section: u64) -> Result<u64, Error> {
+        self.prune_guard(section, false)?;
+        match self.blobs.get(&section) {
+            Some(blob) => Ok(blob.size().await),
+            None => Ok(0),
+        }
+    }
+
+    /// Gets the highest existing section ID, or None if no sections exist.
+    pub fn max_section_id(&self) -> Option<u64> {
+        self.blobs.keys().max().copied()
+    }
+
+    /// Truncates a specific section to the given offset.
+    ///
+    /// This removes any data in the section beyond the specified offset.
+    /// The offset is in journal offset units (aligned to ITEM_ALIGNMENT).
+    pub async fn truncate_section(&mut self, section: u64, offset: u32) -> Result<(), Error> {
```

Comment:
Rename to `truncate` and fix comment (ensures journal exists up to section/offset)